### PR TITLE
Update Logentries to use new Create/Update pattern

### PR DESCRIPTION
**Pull request by mccurdyc**
## Proposed Change(s)

* Uses a factory pattern for Creates and Updates.
* Adds test for non-exported API in `_test.go` file.
* Moves exported API test to `_integration_test.go` file.

## Validation

```
$ make test
```

```
git clone git@github.com:mccurdyc/cli.git /tmp/cli && \
(
    cd /tmp/cli && \
    git checkout mccurdyc/logging-test-logentries && \
    make test && \
    rm -rf /tmp/cli \
)
```

### CODE REVIEW

File: pkg/logging/logentries/create.go [hunk 1]
```diff
@@ -15,11 +15,19 @@ import (
 type CreateCommand struct {
 	common.Base
 	manifest manifest.Data
-	Input    fastly.CreateLogentriesInput
 
-	// We must store all of the boolean flags seperatly to the input structure
-	// so they can be casted to go-fastly's custom `Compatibool` type later.
-	useTLS bool
```
Comment (mccurdyc) [inline pkg/logging/logentries/create.go:22]: Yay! No more this.

File: pkg/logging/logentries/logentries_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,389 @@
+package logentries_test
```
Comment (mccurdyc) [inline pkg/logging/logentries/logentries_integration_test.go:1]: Straight `mv logentries_test.go logentries_integration_test.go`

File: pkg/logging/logentries/update.go [hunk 3]
```diff
@@ -85,10 +90,6 @@ func (c *UpdateCommand) Exec(in io.Reader, out io.Writer) error {
 		input.NewName = c.NewName.Value
 	}
 
-	if c.NewName.Valid {
```
Comment (mccurdyc) [inline pkg/logging/logentries/update.go:88]: Removed a duplicate set.

Comment (phamann) [reply pkg/logging/logentries/update.go:88]: Nice catch!

**Comment by mccurdyc** [comment]
Force-pushed rebasing master.

**Pull request merged**